invoices/invoiceregistry.go

```diff
@@ -1499,9 +1506,10 @@ func (i *InvoiceRegistry) SubscribeNotifications(
 		addIndex:        addIndex,
 		settleIndex:     settleIndex,
 		invoiceSubscriptionKit: invoiceSubscriptionKit{
-			inv:        i,
-			ntfnQueue:  queue.NewConcurrentQueue(20),
-			cancelChan: make(chan struct{}),
+			subscriptionCancels: i.subscriptionCancels,
```

Comment:
I'm not sure I understand this change. Won't this effectively result in the exact same channels being referenced as before? Just not through a pointer anymore? I'm not sure how this should prevent deadlocks?

Reply:
Ideally we have a one-to-many relationship between `InvoiceRegistry` and all the clients, yet putting a whole `InvoiceRegistry` back to each client creates a cyclic reference. The `InvoiceRegistry` has a map of client pointers, and each client has a pointer to the `InvoiceRegistry`. I guess it's just not a healthy pattern IMO. 

Initially I thought it would introduce deadlocks when a client acquires a lock, do something then the `InvoiceRegistry` also acquires a lock, we would have trouble. Turns out we won't. I guess when it locks, only the pointers are locked, not the data the pointers are pointing to.

Reply:
Ah, so it's not really about the two variables/channels in question but there being a cyclic pointer issue in general. Got it.